lms/static/scripts/frontend_apps/components/test/BookPicker-test.js

```diff
@@ -0,0 +1,142 @@
+import { mount } from 'enzyme';
+import { createElement } from 'preact';
+import { act } from 'preact/test-utils';
+
+import mockImportedComponents from '../../../test-util/mock-imported-components';
+import { waitForElement } from '../../../test-util/wait';
+import * as bookData from '../../utils/vitalsource-sample-data';
+import BookPicker, { $imports } from '../BookPicker';
+
+describe('BookPicker', () => {
+  let fakeAPI;
+
+  const renderBookPicker = (props = {}) =>
+    mount(<BookPicker authToken="dummy-auth-token" {...props} />);
+
+  beforeEach(() => {
+    fakeAPI = {
+      fetchBooks: sinon.stub().resolves(bookData.bookList),
+      fetchChapters: sinon
+        .stub()
+        .callsFake(async (authToken, bookID) => bookData.chapterData[bookID]),
+    };
+
+    $imports.$mock(mockImportedComponents());
+    $imports.$restore({
+      './Dialog': true,
+    });
+    $imports.$mock({
+      '../utils/api': fakeAPI,
+    });
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  const selectBook = wrapper => {
+    const bookList = wrapper.find('BookList');
+    const book = bookList.prop('books')[0];
+    act(() => {
+      bookList.prop('onSelectBook')(book);
+    });
+    wrapper.update();
+    return book;
+  };
```

Comment:
In the manner that `clickSelectButton` below is crystal-clear naming, would this more accurately be `selectFirstBook`? It might help a dev scan more quickly here.

Reply:
Hmm... it might be useful to have tests that select other books in future. I think I can achieve the desired result by replacing the hard-coded `0` with an `index` which is a parameter with a default value of `0`.